src/settings/global_user.rs

```diff
@@ -86,40 +166,97 @@ impl GlobalUser {
-                let wrangler_config_msg = styles::highlight("`wrangler config`");
-                let vars_msg = styles::url("https://developers.cloudflare.com/workers/tooling/wrangler/configuration/#using-environment-variables");
-                let msg = format!(
-                    "{} Your authentication details are improperly configured.\nPlease run {}, {}, or visit\n{}\nfor info on configuring with environment variables",
-                    emoji::WARN,
-                    wrangler_login_msg,
-                    wrangler_config_msg,
-                    vars_msg
-                );
-                log::info!("{:?}", config);
-                anyhow::bail!(msg)
-            }
+        // Get all the possible authentication methods
+        let api_token = config.get_str("api_token");
+        let oauth_token = config.get_str("oauth_token");
+        let email = config.get_str("email");
+        let api_key = config.get_str("api_key");
+        let refresh_token = config.get_str("refresh_token");
+        let expiration_time = config.get_str("expiration_time");
+
+        // The only cases that are not allowed are:
+        //      1) (partial or complete) OAuth token + API token
+        //      2) (partial or complete) OAuth token + Global API key (partial or complete)
+        //      3) Invalid authentication methods (e.g. partial Oauth token, partial Global API key, and empty configuration file + no environment variables)
+        // API token has priority over global API key both in environment variables and in configuration file
+        if (api_token.is_ok()
+            && (oauth_token.is_ok() || refresh_token.is_ok() || expiration_time.is_ok()))
+            || ((oauth_token.is_ok() || refresh_token.is_ok() || expiration_time.is_ok())
+                && (email.is_ok() || api_key.is_ok()))
+        {
+            let error_info = "\nMore than one authentication method (e.g. API token and OAuth token, or OAuth token and Global API key) has been found in the configuration file. Please use only one.";
+            Self::show_config_err_info(Some(error_info.to_string()), config)
+        } else if api_token.is_ok() {
+            Ok(Self::ApiTokenAuth {
+                api_token: api_token.expect("Failed to read API token"),
+            })
+        } else if email.is_ok() && api_key.is_ok() {
+            Ok(Self::GlobalKeyAuth {
+                email: email.expect("Failed to read email"),
+                api_key: api_key.expect("Failed to read api_key"),
+            })
+        } else if oauth_token.is_ok() && refresh_token.is_ok() && expiration_time.is_ok() {
+            Ok(Self::OAuthTokenAuth {
+                oauth_token: oauth_token.expect("Failed to read OAuth token"),
+                refresh_token: refresh_token.expect("Failed to read OAuth refresh token"),
+                expiration_time: expiration_time
+                    .expect("Failed to read access token expiration time"),
```

Comment:
I think all of these `expect` calls can just be `unwrap`, since you've already attested the are `Ok` in the if checks